src/perfEvents_linux.cpp

```diff
@@ -937,9 +938,9 @@ int PerfEvents::walk(int tid, void* ucontext, const void** callchain, int max_de
     event->unlock();
 
     if (_cstack == CSTACK_FP) {
-        depth += StackWalker::walkFP(ucontext, callchain + depth, max_depth - depth, java_ctx);
+        depth += StackWalker::walkFP(ucontext, callchain + depth, max_depth - depth, java_ctx, truncated);
     } else if (_cstack == CSTACK_DWARF) {
-        depth += StackWalker::walkDwarf(ucontext, callchain + depth, max_depth - depth, java_ctx);
+        depth += StackWalker::walkDwarf(ucontext, callchain + depth, max_depth - depth, java_ctx, truncated);
```

Comment:
`truncated` flag is not really meaningful with `fp` and `dwarf` stack walking. It can be even misleading, since it refers to the native part of a stack only. `AsyncGetCallTrace` may return complete or truncated stacktrace, and profiler has no means to find this out.

Whether a stacktrace is truncated or not, can be only determined with VMStructs-based stack walking.

Reply:
Ok, the idea is to have an indication when we didn't walk up to root. Either because some error or because we reached the maximum number of frames we are allowed to walk (eg. because of the stackdepth argument).
Does it make sense?

Reply:
But that's not what the current implementation does. In this PR, `truncated` flag can be set for stacks fully unwound to the bottommost `Thread.run` or vice versa: a partial stack can be returned with `truncated = false`, because all stack unwinding modes except `vm` rely on `AsyncGetCallTrace`. I'd suggest not to confuse people and to leave `fp` and `dwarf` modes intact.